pkg/controller/volume/persistentvolume/pv_controller_base.go

```diff
@@ -126,10 +127,19 @@ func NewController(p ControllerParameters) (*PersistentVolumeController, error)
 	controller.classLister = p.ClassInformer.Lister()
 	controller.classListerSynced = p.ClassInformer.Informer().HasSynced
 	controller.podLister = p.PodInformer.Lister()
+	controller.podIndexer = p.PodInformer.Informer().GetIndexer()
 	controller.podListerSynced = p.PodInformer.Informer().HasSynced
 	controller.NodeLister = p.NodeInformer.Lister()
 	controller.NodeListerSynced = p.NodeInformer.Informer().HasSynced
 
+	// This custom indexer will index pods by its PVC keys. Then we don't need
+	// to iterate all pods every time to find pods which reference given PVC.
+	if err := controller.podIndexer.AddIndexers(cache.Indexers{
+		pvcKeyIndex: indexByPVCKey,
+	}); err != nil {
+		klog.Warningf("adding indexer got %v", err)
```

Comment:
Why is the error not propagated to the caller?

Reply:
I copied this snippet of code from ADC. The error wouldn't happen in practice but it makes code smell better if we propagate the error to the caller. I'll fix this.